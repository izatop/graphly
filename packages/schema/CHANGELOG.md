# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.6.5](https://github.com/graphly/graphly/compare/v0.6.4...v0.6.5) (2021-04-14)


### Bug Fixes

* update deps ([e99fa4b](https://github.com/graphly/graphly/commit/e99fa4beeb580e78aaa6d44c65b16d0ed7560e87))





## [0.6.4](https://github.com/graphly/graphly/compare/v0.6.3...v0.6.4) (2021-03-25)

**Note:** Version bump only for package @graphly/schema





## [0.6.3](https://github.com/graphly/graphly/compare/v0.6.2...v0.6.3) (2021-03-18)


### Bug Fixes

* update deps ([2126c5b](https://github.com/graphly/graphly/commit/2126c5b810828ee664cdaa07d8947ab04aaa4924))





## [0.6.2](https://github.com/graphly/graphly/compare/v0.6.1...v0.6.2) (2021-02-25)


### Bug Fixes

* safe DateType wrap ([3c290ce](https://github.com/graphly/graphly/commit/3c290ce62b74e7efa8db276fe5c404b20f8858ea))





## [0.6.1](https://github.com/graphly/graphly/compare/v0.6.0...v0.6.1) (2021-02-09)


### Bug Fixes

* update deps ([32e419f](https://github.com/graphly/graphly/commit/32e419fa87b6326d687c494d3f6189aaa9962679))





# [0.6.0](https://github.com/graphly/graphly/compare/v0.5.11...v0.6.0) (2021-01-25)

**Note:** Version bump only for package @graphly/schema





## [0.5.10](https://github.com/graphly/graphly/compare/v0.5.9...v0.5.10) (2021-01-22)

**Note:** Version bump only for package @graphly/schema





## [0.5.9](https://github.com/graphly/graphly/compare/v0.5.8...v0.5.9) (2021-01-22)


### Bug Fixes

* Mutate the subscription source by the GraphQLResolveInfo object ([fb0dac9](https://github.com/graphly/graphly/commit/fb0dac95e2cb5edadc0bfdeb8e66861671df2dba))





## [0.5.8](https://github.com/graphly/graphly/compare/v0.5.7...v0.5.8) (2021-01-20)


### Bug Fixes

* update tests ([ea877d3](https://github.com/graphly/graphly/commit/ea877d36c59e81ea984affa43213d3e75c173f74))





## [0.5.6](https://github.com/graphly/graphly/compare/v0.5.5...v0.5.6) (2020-10-29)


### Bug Fixes

* update deps ([961b557](https://github.com/graphly/graphly/commit/961b557fd51c9a486727f67e8518c54bfce76772))
* update deps ([7743538](https://github.com/graphly/graphly/commit/7743538f1d158625a898a3cb2d2ef5703c779aff))





## [0.5.4](https://github.com/graphly/graphly/compare/v0.5.3...v0.5.4) (2020-09-19)


### Bug Fixes

* update deps ([8669c10](https://github.com/graphly/graphly/commit/8669c100fa1abd9a9bb5fafd91472df076be4062))





## [0.5.3](https://github.com/graphly/graphly/compare/v0.5.2...v0.5.3) (2020-09-09)


### Bug Fixes

* update deps ([360f7bf](https://github.com/graphly/graphly/commit/360f7bf0435a09c6f9e157927c869b20a6239884))





## [0.5.1](https://github.com/graphly/graphly/compare/v0.5.0...v0.5.1) (2020-08-08)


### Bug Fixes

* update deps ([6f725b7](https://github.com/graphly/graphly/commit/6f725b7e79e0d7d47b6e48098cb23781c76a96fb))





# [0.5.0](https://github.com/graphly/graphly/compare/v0.4.19...v0.5.0) (2020-07-13)


### Bug Fixes

* parse, tests ([d388f8c](https://github.com/graphly/graphly/commit/d388f8cdb3b7c2ff5d1ff17915c288a0aa25189d))


### Features

* phone number type ([8f6961a](https://github.com/graphly/graphly/commit/8f6961ac4b6d8c15e06c4dc53d95e7917859c0b2))
* simple phone validation -> phone npm package ([6918fc5](https://github.com/graphly/graphly/commit/6918fc5cfab15a0d59f6fda721c828a628820131))





## [0.4.19](https://github.com/graphly/graphly/compare/v0.4.18...v0.4.19) (2020-07-08)


### Bug Fixes

* domain name regex ([17d4c79](https://github.com/graphly/graphly/commit/17d4c79964620d251260b9258cdf3b45dfb038b9))





## [0.4.18](https://github.com/graphly/graphly/compare/v0.4.17...v0.4.18) (2020-07-05)


### Bug Fixes

* default values ([2c749e2](https://github.com/graphly/graphly/commit/2c749e2b4641837fa387f767dcf46e7e5f16df9c))





## [0.4.17](https://github.com/graphly/graphly/compare/v0.4.16...v0.4.17) (2020-07-05)


### Bug Fixes

* default values ([401291b](https://github.com/graphly/graphly/commit/401291babc6d4417e772aa036e0f2a6c70c11ae4))





## [0.4.16](https://github.com/graphly/graphly/compare/v0.4.15...v0.4.16) (2020-07-05)


### Bug Fixes

* root dir(s) ([f7bccc2](https://github.com/graphly/graphly/commit/f7bccc254b639ee13401c20a4bfc77b5fcd9b316))





## [0.4.15](https://github.com/graphly/graphly/compare/v0.4.14...v0.4.15) (2020-07-05)


### Bug Fixes

* extra types ([453c416](https://github.com/graphly/graphly/commit/453c4162944de88a2d16b696579dc0f2cf8cffa4))





## [0.4.13](https://github.com/graphly/graphly/compare/v0.4.12...v0.4.13) (2020-07-04)

**Note:** Version bump only for package @graphly/schema





## [0.4.12](https://github.com/graphly/graphly/compare/v0.4.11...v0.4.12) (2020-07-02)


### Bug Fixes

* update deps, factory ([b7833fb](https://github.com/graphly/graphly/commit/b7833fb1bc874bacaa1cd77116128629dc1f81d0))





## [0.4.11](https://github.com/graphly/graphly/compare/v0.4.10...v0.4.11) (2020-07-01)


### Bug Fixes

* bug ([278e83d](https://github.com/graphly/graphly/commit/278e83de96da7edbe9bc23fbc405f04100383c1d))





## [0.4.10](https://github.com/graphly/graphly/compare/v0.4.9...v0.4.10) (2020-07-01)


### Bug Fixes

* scalar types ([45aad63](https://github.com/graphly/graphly/commit/45aad63558b2faa465390a5a1d7e31341c867719))





## [0.4.9](https://github.com/graphly/graphly/compare/v0.4.8...v0.4.9) (2020-06-30)


### Bug Fixes

* supports email, domain and ip4 ([49a59a6](https://github.com/graphly/graphly/commit/49a59a648ac8758b2d1b4716222fb6bd660a9878))





## [0.4.8](https://github.com/graphly/graphly/compare/v0.4.7...v0.4.8) (2020-06-24)

**Note:** Version bump only for package @graphly/schema





## [0.4.7](https://github.com/graphly/graphly/compare/v0.4.6...v0.4.7) (2020-06-22)


### Bug Fixes

* update deps ([0c3774a](https://github.com/graphly/graphly/commit/0c3774a09bee309835b5bfad823aace93be695b6))





## [0.4.6](https://github.com/graphly/graphly/compare/v0.4.5...v0.4.6) (2020-06-11)


### Bug Fixes

* cache intersection ([64ff189](https://github.com/graphly/graphly/commit/64ff189f4b24518e8c12b11a1558c0f27f5cd603))
* lint ([97559c5](https://github.com/graphly/graphly/commit/97559c59a89aa1f60169085ade2d7a4b68248191))





## [0.4.5](https://github.com/graphly/graphly/compare/v0.4.4...v0.4.5) (2020-06-09)


### Bug Fixes

* prototype chain ([76c51d1](https://github.com/graphly/graphly/commit/76c51d1f1726adfd02af3acb4368d0db4e3fbbe3))





## [0.4.2](https://github.com/graphly/graphly/compare/v0.4.1...v0.4.2) (2020-05-12)


### Bug Fixes

* object rest order ([4fd736f](https://github.com/graphly/graphly/commit/4fd736f00eef592d9737cae00ae7c0c86d4bdf7f))
* skip wornings ([19ed050](https://github.com/graphly/graphly/commit/19ed0504b6c88750b38a541e99cee76c9704f20e))





## [0.4.1](https://github.com/graphly/graphly/compare/v0.4.0...v0.4.1) (2020-04-30)


### Bug Fixes

* watch mode, update deps ([193569d](https://github.com/graphly/graphly/commit/193569d78b7c9e68f2f04caca318153a681c996b))





# [0.4.0](https://github.com/graphly/graphly/compare/v0.3.5...v0.4.0) (2020-04-24)


### Features

* scope craete factory ([f55f363](https://github.com/graphly/graphly/commit/f55f363e08909bee53e2e92b0535b025169e27d3))





## [0.3.5](https://github.com/graphly/graphly/compare/v0.3.4...v0.3.5) (2020-04-24)


### Bug Fixes

* cli and gql transform ([026c622](https://github.com/graphly/graphly/commit/026c622222f1fbee1a97e087e9b2cc56ab3e3ce6))





## [0.3.4](https://github.com/graphly/graphly/compare/v0.3.3...v0.3.4) (2020-04-23)


### Bug Fixes

* resolve a container ([362d66d](https://github.com/graphly/graphly/commit/362d66d0e24526545be7a2582b417c53eb324f45))





# [0.3.0](https://github.com/graphly/graphly/compare/v0.2.17...v0.3.0) (2020-04-13)


### Features

* typedoc 0.17, use json reflection ([7563766](https://github.com/graphly/graphly/commit/756376610c9b573acbe7d581c9ae4efe8103a5a4))





## [0.2.17](https://github.com/graphly/graphly/compare/v0.2.16...v0.2.17) (2020-04-07)


### Bug Fixes

* travis-ci ([961d322](https://github.com/graphly/graphly/commit/961d322558327bddd80450172583119fb6cc5b50))





## [0.2.16](https://github.com/graphly/graphly/compare/v0.2.15...v0.2.16) (2020-04-07)


### Bug Fixes

* update deps ([11ebdf7](https://github.com/graphly/graphly/commit/11ebdf75403e5ee0fd449de1560d4b25f876a664))
* update deps ([fcb7505](https://github.com/graphly/graphly/commit/fcb7505420631e2fe2567cf6f415b122a4576038))





## [0.2.14](https://github.com/graphly/graphly/compare/v0.2.13...v0.2.14) (2019-11-25)


### Bug Fixes

* update deps ([c12cb4f](https://github.com/graphly/graphly/commit/c12cb4fe0eb76697731c3c2e2c35d1b512b88257))





## [0.2.13](https://github.com/graphly/graphly/compare/v0.2.12...v0.2.13) (2019-11-18)


### Bug Fixes

* update deps ([0842962](https://github.com/graphly/graphly/commit/08429624fe0565adfc56ce271cc10b0251c5c156))





## [0.2.11](https://github.com/graphly/graphly/compare/v0.2.10...v0.2.11) (2019-10-23)


### Bug Fixes

* update deps ([239ac51](https://github.com/graphly/graphly/commit/239ac51e740158e02b1717d721dcb82c0de80462))





## [0.2.10](https://github.com/graphly/graphly/compare/v0.2.9...v0.2.10) (2019-10-14)


### Bug Fixes

* ObjectType field resolution fix ([cbb9664](https://github.com/graphly/graphly/commit/cbb9664bc2f529549c905b0a1d8f66dea4c6626b))





## [0.2.8](https://github.com/graphly/graphly/compare/v0.2.7...v0.2.8) (2019-10-10)


### Bug Fixes

* update collision in interface name generation ([1ce45e7](https://github.com/graphly/graphly/commit/1ce45e7d86e98bcf190a092f1e3f0f6756603a6b))





## [0.2.7](https://github.com/graphly/graphly/compare/v0.2.6...v0.2.7) (2019-10-09)


### Bug Fixes

* merge ci updates and fix README ([8b6ca06](https://github.com/graphly/graphly/commit/8b6ca06c0411cd41687a7356dc5e018cb0725d16))
* return types for async resolvers ([35afcb2](https://github.com/graphly/graphly/commit/35afcb255f00481f4bcf1c6b271f9c755a4ef713))





## [0.2.6](https://github.com/graphly/graphly/compare/v0.2.5...v0.2.6) (2019-10-07)


### Bug Fixes

* update deps ([f35a77f](https://github.com/graphly/graphly/commit/f35a77f))





## [0.2.5](https://github.com/graphly/graphly/compare/v0.2.4...v0.2.5) (2019-09-19)


### Bug Fixes

* update README and skip plain interfaces ([a9ecde9](https://github.com/graphly/graphly/commit/a9ecde9))





## [0.2.2](https://github.com/graphly/graphly/compare/v0.2.1...v0.2.2) (2019-09-15)


### Bug Fixes

* improve a return type resolution ([48d9789](https://github.com/graphly/graphly/commit/48d9789))





# [0.2.0](https://github.com/graphly/graphly/compare/v0.1.4...v0.2.0) (2019-09-15)


### Features

* cli package ([1b20e04](https://github.com/graphly/graphly/commit/1b20e04))
