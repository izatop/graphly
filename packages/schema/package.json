{
    "name": "@graphly/schema",
    "version": "0.8.0",
    "keywords": [
        "graphql",
        "typescript"
    ],
    "author": {
        "name": "Artur Bier",
        "email": "izatop@gmail.com"
    },
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "files": [
        "dist/"
    ],
    "description": "Create GraphQL schema/source from typemap cache.",
    "repository": "https://github.com/graphly/graphly",
    "scripts": {
        "clean": "yarn build:clean",
        "watch": "yarn build:watch",
        "build": "yarn build:clean && tsc",
        "build:clean": "rimraf dist tsconfig.tsbuildinfo",
        "build:watch": "yarn build --watch"
    },
    "dependencies": {
        "@sirian/assert": "^1.1.21",
        "@sirian/common": "^9.5.3",
        "@sirian/decorators": "^2.1.3",
        "@types/phone": "^2.4.1",
        "graphql": "^16.2.0",
        "phone": "^3.1.10"
    },
    "publishConfig": {
        "access": "public"
    }
}
