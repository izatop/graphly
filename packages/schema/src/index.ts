import * as Scalars from "./Scalars";

export * from "./Project";
export * from "./Type";
export {Scalars};
