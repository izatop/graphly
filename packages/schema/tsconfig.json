{
    "extends": "../../tsconfig.lib.json",
    "compilerOptions": {
        "rootDir": "src",
        "outDir": "dist",
        "incremental": true
    },
    "include": [
        "src"
    ]
}
