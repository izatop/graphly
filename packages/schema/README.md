# @graphly/schema

Create GraphQL schema/source from typemap cache.

This package is a part of [Graphly](https://github.com/graphly/graphly) project and uses by
`@graphly/type` to convert TypeScript classes to GraphQL schema.

## Install

`yarn add @graphly/schema`

## Usage

```typescript
import {Project} from "@graphly/schema";
import {MySchema} from "./Schema/MySchema.ts"

async function main() {
    const project = await Project.from(MySchema.getSchemaLocation());
    
    // convert project to GraphQL format
    console.log(project.toGraphQL());
    
    // convert project to GraphQLSchema
    console.log(project.toSchema());
}

main();
```

## Links

Read more about [this project](https://github.com/graphly/graphly).

## License
MIT
