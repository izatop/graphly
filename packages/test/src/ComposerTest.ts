import {Project} from "@graphly/schema";
import {GraphQLSchema} from "graphql";
import {TestSchema} from "./TestSchema";

test("Composer", async () => {
    const project = await Project.from(TestSchema.getSchemaLocation());
    expect(project.toGraphQL()).toMatchSnapshot();
    expect(project.toSchema()).toBeInstanceOf(GraphQLSchema);
});
