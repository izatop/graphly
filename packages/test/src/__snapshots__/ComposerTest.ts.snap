// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Composer 1`] = `
"scalar DateTime
scalar DomainName
scalar EmailAddress
scalar IPv4Address
scalar Object

input TodoChecklistInput {
  solved: Boolean
  text: String!
}

input TodoInput {
  checklist: [TodoChecklistInput!]
  deadlineAt: DateTime
  description: String
  flag: TodoFlag
  solved: Boolean
  title: String
}

input TodoSearchInput {
  limit: Int
  offset: Int
  solved: Boolean
}

type TodoMutation {
  add(todo: TodoInput!): Todo!
  delete(id: [Int!]!): [Boolean!]!
  update(id: Int!, todo: TodoInput!): Todo
}

type Todo {
  checklist: [TodoChecklist!]!
  deadlineAt: DateTime
  description: String
  flag: TodoFlag
  id: Int!
  solved: Boolean!
  title: String!
  code: String!
  timestamp: Int!
}

type TodoChecklist {
  solved: Boolean
  text: String!
}

enum TodoFlag {
  NONE
  PRIVATE
  PUBLIC
}

type IPageableTodo {
  count: Int!
  limit: Int!
  node: [Todo!]!
  offset: Int!
}

type TodoQuery {
  count(solved: Boolean = true): Int!
  search(filter: TodoSearchInput): IPageableTodo!
  searchAll(limit: Float = 10, offset: Float = 0): IPageableTodo!
  todo(id: Int!): Todo
}

type TestMutation {
  todo: TodoMutation!
}

type TestQuery {
  domain: DomainName!
  email: EmailAddress!
  ipv4: IPv4Address!
  optional: Float
  random: Float!
  todo: TodoQuery!
  timestamp: Float!
  config: Boolean!
  hello: String!
}

type TestSubscription {
  onTodoAdded: Todo!
  onTodoDelete: Todo!
  onTodoUpdate: Todo!
}

schema {
  mutation: TestMutation
  query: TestQuery
  subscription: TestSubscription
}"
`;
